nkms/crypto/signature.py

```diff
@@ -1,57 +1,59 @@
 from nkms.crypto import api as API
-from nkms.keystore.keypairs import PublicKey
+from umbral.keys import UmbralPublicKey
+from cryptography.hazmat.primitives.asymmetric.utils import decode_dss_signature, encode_dss_signature
 
 
-class Signature(bytes):
+class Signature(object):
     """
     The Signature object allows signatures to be made and verified.
     """
-    _EXPECTED_LENGTH = 65
+    _EXPECTED_LENGTH = 64  # With secp256k1 and BLAKE2b(64).
 
-    def __init__(self, sig_as_bytes: bytes=None, v: int=None, r: int=None, s: int=None):
-        """
-        Initializes a Signature object.
-
-        :param v: V param of signature
-        :param r: R param of signature
-        :param s: S param of signature
-
-        :return: Signature object
-        """
-        if sig_as_bytes and any((v, r, s)):
-            raise ValueError("Pass *either* sig_as_bytes *or* v, r, and s - don't pass both.")
-        if not any ((sig_as_bytes, v, r, s)):
-            raise ValueError("Pass either sig_as_bytes or v, r, and s.")
-
-        if sig_as_bytes:
-            if not len(sig_as_bytes) == self._EXPECTED_LENGTH:
-                raise ValueError("{} must be {} bytes.".format(self.__class__.__name__, self._EXPECTED_LENGTH))
-            v, r, s = API.ecdsa_load_sig(sig_as_bytes)
-
-        self._v = v
-        self._r = r
-        self._s = s
+    def __init__(self, r: int, s: int):
+        self.r = r
+        self.s = s
 
     def __repr__(self):
-        return "{} v{}: {} - {}".format(__class__.__name__, self._v, self._r, self._s)
+        return "ECDSA Signature: {}".format(bytes(self).hex()[:15])
```

Comment:
Why just the first 15 bytes?

Reply:
I guess, to just identify the signature in a readable way rather than to print the full signature

Reply:
I like to keep `__repr__` reasonably short so that it can be skimmed in logs.  If you think there's a need to have the whole sig, I don't have a huge problem with that.

Reply:
No no, just wanted to know the reason. Plus, 15 seems weird since it encodes 7.5 bytes

Reply:
Heh... well, shall I change it?

Reply:
Fine for me.. It's clear that repr isn't for serialization, and that is more clear with 7.5 bytes!

Reply:
✔️